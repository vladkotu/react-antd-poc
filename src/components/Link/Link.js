import React from 'react'
import PropTypes from 'prop-types'
import { PlusOutlined, EditOutlined } from '@ant-design/icons'
import './Link.css'

function Link({ children, add, edit, ...props }) {
  return (
    <a {...props}>
      {add && <PlusOutlined className='Link-icon' />}
      {edit && <EditOutlined className='Link-icon' />}
      {children}
    </a>
  )
}

Link.propTypes = {
  add: PropTypes.bool,
  edit: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

export default Link
