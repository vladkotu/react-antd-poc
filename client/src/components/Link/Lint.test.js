import React from 'react'
import { render } from '@testing-library/react'
import Link from './Link'

describe('Link', () => {
  test('renders a tag', () => {
    const testHref = '#'
    const { getByText, container } = render(
      <Link href={testHref}>Link text</Link>
    )
    const link = getByText('Link text')
    expect(link.closest('a')).toBeInTheDocument()
    expect(link.closest('a')).toHaveAttribute('href', testHref)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders link without logo by default', () => {
    const { queryByRole } = render(<Link href='#'>Link text</Link>)
    const icon = queryByRole('img')
    expect(icon).toBeNull()
  })

  test('able to render edit link', () => {
    const icons = [
      ['add', 'plus'],
      ['edit', 'edit'],
    ]
    for (let [flag, role] of icons) {
      const { getByLabelText } = render(
        <Link {...{ [flag]: true }} href='#'>
          Link text
        </Link>
      )
      const img = getByLabelText(role)
      expect(img).toBeInTheDocument()
    }
  })
})
