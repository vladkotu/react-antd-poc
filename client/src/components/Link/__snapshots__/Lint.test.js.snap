// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders a tag 1`] = `
<a
  href="#"
>
  Link text
</a>
`;
